nexus/transport.go

```diff
@@ -0,0 +1,707 @@
+package nexus
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"maps"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/google/uuid"
+)
+
+const getResultContextPadding = time.Second * 5
+
+var errOperationWaitTimeout = errors.New("operation wait timeout")
+
+type (
+	// Transport is the low-level abstraction used by [Client] and [OperationHandle] to make network calls.
+	//
+	// NOTE: Experimental
+	Transport interface {
```

Comment:
I think this needs a `Close()`. This brings up a philosophical discussion where we have to ask whether `Client` needs a `Close` and does it call `Transport.Close`. My opinion is, if you are going to keep the create-client-and-transport helper, you need a `Client.Close` that only invokes `Transport.Close` if it was created with the client by our code, otherwise it's a no-op (could do counters, but that's too much magic). If you are not going to keep the create-client-and-transport helper and instead require two-step instantiation, you don't need a `Client.Close` at all. I lean the latter since the Client returned from Temporal SDK doesn't need to be `Close`d (because it's outer client is responsible for that).